repository/src/test/java/org/alfresco/repo/download/DownloadServiceIntegrationTest.java

```diff
@@ -207,28 +218,55 @@ public class DownloadServiceIntegrationTest
         NodeRef COMPANY_HOME = repositoryHelper.getCompanyHome();
         
         // Create some static test content
-       rootFolder = testNodes.createNode(COMPANY_HOME, "rootFolder", ContentModel.TYPE_FOLDER, AuthenticationUtil.getAdminUserName());
-       allEntries.add("rootFolder/");
+        rootFolder = testNodes.createNode(COMPANY_HOME, "rootFolder", ContentModel.TYPE_FOLDER, AuthenticationUtil.getAdminUserName());
+        allEntries.add("rootFolder/");
+
+        TRANSACTION_HELPER.doInTransaction(new RetryingTransactionCallback<Void>()
+        {
+            public Void execute() throws Throwable
+            {
+                org.alfresco.service.cmr.rule.Rule parentRule = new org.alfresco.service.cmr.rule.Rule();
+                parentRule.setRuleTypes(Collections.singletonList(RuleType.INBOUND));
+                parentRule.setTitle("DownloadServiceIntegrationTest" + GUID.generate());
+                parentRule.setDescription("Add Classifiable");
+                Action action = ACTION_SERVICE.createAction(AddFeaturesActionExecuter.NAME);
+                action.setParameterValue(AddFeaturesActionExecuter.PARAM_ASPECT_NAME, ContentModel.ASPECT_CLASSIFIABLE);
+                parentRule.setAction(action);
+                parentRule.applyToChildren(true);
+
+                RULE_SERVICE.saveRule(rootFolder, parentRule);
+
+                return null;
+            }
+        }, false, true);
 
        rootFile = testNodes.createNodeWithTextContent(COMPANY_HOME, "rootFile.txt", ContentModel.TYPE_CONTENT, AuthenticationUtil.getAdminUserName(), "Root file content");
        allEntries.add("rootFile.txt");
-       
-       testNodes.createNodeWithTextContent(rootFolder, "level1File.txt", ContentModel.TYPE_CONTENT, AuthenticationUtil.getAdminUserName(), "Level 1 file content");
+
+       NodeRef createdNode;
+
+       createdNode = testNodes.createNodeWithTextContent(rootFolder, "level1File.txt", ContentModel.TYPE_CONTENT, AuthenticationUtil.getAdminUserName(), "Level 1 file content");

```

Comment:
You could have `NodeRef` at the start of this line instead?

Reply:
I can but, because `createdNode` is a variable that going to be rewritten later on, it was a way to make it more visible.